mattermostLatest: init at 10.4.1; mattermost: 9.11.6 -> 9.11.7

numinit opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Create `mattermostLatest` to pull in the latest Mattermost. This works fine with all the NixOS tests, but that won't be PRed here. Instead, we will do it in #208181 which will not be backported. If people choose the latest Mattermost, they are accepting a little less stability (in general, but especially for now).

Depends-on: #373085

Backport label this since it affects the autoupdate.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

numinit committed:
mattermostLatest: init at 10.3.1

numinit commented:
@fsagbuya No changes were necessary for this to get it working

fsagbuya commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 373113`

fsagbuya commented:
LGTM. :)

numinit commented:
Looks like an empty result? In any case I think it's just a `nix-build -A mattermostLatest`

fsagbuya commented:
> Looks like an empty result? In any case I think it's just a `nix-build -A mattermostLatest`

```
nix-build -A mattermostLatest && ls -lhLS result/
/nix/store/zspzwzcf4mvq44szgchvvah3xl1qzb0k-mattermost-10.3.1
total 52K
dr-xr-xr-x 8 root root  28K Jan  1  1970 client
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 bin
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 config
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 fonts
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 i18n
dr-xr-xr-x 3 root root 4.0K Jan  1  1970 templates
```

Package built. Can `nixosTests.mattermost` be applied to this as well?

numinit commented:
It will be in #208181, but I manually replaced pkgs.mattermost with pkgs.mattermostLatest and they all passed. #208181 changes the module enough (in a backwards compatible way, but a way that affects the tests nonetheless) that it's probably not worth casing out the tests in this PR.

I've not pushed up the changes just yet but it starts with Mattermost ESR and then switches to a config specialization that switches to mattermostLatest to make sure migrations work.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2193

numinit commented:
Should be good to merge. We should try to prioritize it since Mattermost updates once a month.

@drupol do you mind taking a look? :-)

drupol commented on pkgs/by-name/ma/mattermost/package.nix:
```diff
@@ -8,21 +8,31 @@
   src = fetchFromGitHub {
     owner = "mattermost";
     repo = "mattermost";
     rev = "v${version}";
```
Can we use tag here?

numinit replied:
Oh, that's brand new, good idea!

numinit committed:
mattermostLatest: use 10.4.x branch

numinit commented:
@drupol Out of curiosity does the NixOS 24.11 branch support that?

drupol commented:
Yes: https://github.com/NixOS/nixpkgs/pull/362113

numinit committed:
mattermostLatest: 10.3.1 -> 10.4.1

Add a lockfileOverlay option that lets us make changes to the package
lockfile if it's broken, such as unlocking packages and letting
npm-lockfile-fix handle them if they are mis-locked.

numinit committed:
mattermost: 9.11.6 -> 9.11.7

numinit commented:
A Mattermost update snuck in!

numinit committed:
mattermostLatest: track 10.x instead of 10.4.x

We get a new 10.x release every month, so just track latest.
